### PR TITLE
Add TaggedMetricName, Timer/Meter extensions that support metric-leve…

### PR DESCRIPTION
**Pull request by hanwavefront**
…l TimeUnits, additional configuration options for reporter

TODO: 
- Timer/Histogram extensions that support clearing
- String -> TaggedMetricName cache in reporter

### CODE REVIEW

File: src/main/java/com/codahale/metrics/CustomTimeUnitMeter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package com.codahale.metrics;
+
+import com.wavefront.dropwizard.metrics.TaggedMetricName;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link Meter} thta has its own {@link TimeUnit} for rate.
```
Comment (srujann) [inline src/main/java/com/codahale/metrics/CustomTimeUnitMeter.java]: Typo
```suggestion
 * A {@link Meter} that has its own {@link TimeUnit} for rate.
```

Comment (srujann) [inline src/main/java/com/codahale/metrics/CustomTimeUnitMeter.java]: typo in comment.

```suggestion
 * A {@link Meter} that has its own {@link TimeUnit} for rate.
```

File: src/main/java/com/wavefront/dropwizard/metrics/TaggedMetricName.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+   */
+  public static boolean remove(MetricRegistry registry, TaggedMetricName taggedMetricName) {
+    return registry.remove(taggedMetricName.encode());
+  }
+
+  private static Pair<String, String>[] makeTags(Map<String, String> tags) {
+    if (tags == null) {
+      throw new IllegalArgumentException("Tags needs to be specified");
+    }
+    @SuppressWarnings("unchecked")
```
Comment (srujann) [inline src/main/java/com/wavefront/dropwizard/metrics/TaggedMetricName.java:222]: why is this needed?

File: src/main/java/com/codahale/metrics/CustomTimeUnitMeter.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            if (existing instanceof CustomTimeUnitMeter) {
+                return (CustomTimeUnitMeter) existing;
+            } else {
+                throw new IllegalStateException("Existing metric of type: Meter found registered " +
+                        "to metricName: " + metricName);
+            }
+        }
+    }
+
+    public double convertRate(double rate) {
```
Comment (srujann) [inline src/main/java/com/codahale/metrics/CustomTimeUnitMeter.java:46]: java doc for the method?

File: src/main/java/com/wavefront/dropwizard/metrics/TaggedMetricName.java (hunk 2 above)
Comment (hanwavefront) [reply src/main/java/com/wavefront/dropwizard/metrics/TaggedMetricName.java:222]: It suppresses an Unchecked Assignment warning on the Pair[] in the subsequent line. The code that I templated off of did this -- I guess it's a little more performant than building a List and then converting to array

**Comment by hanwavefront** [comment]
The latest commit depends on https://github.com/wavefrontHQ/wavefront-sdk-java/tree/han/update-wfhist-snapshot

**Comment by hanwavefront** [comment]
Not necessary anymore
